### PR TITLE
Add multi-type support on `get_weight()`

### PR DESCRIPTION
**Pull request by datumbox**
Extend the `get_weight()` to handle properly model builders with multiple acceptable weight enums. This is useful for quantization.

cc @datumbox @bjuncek

### CODE REVIEW

File: references/classification/train.py [hunk 1]
```diff
@@ -30,12 +30,12 @@ def train_one_epoch(model, criterion, optimizer, data_loader, device, epoch, arg
     for i, (image, target) in enumerate(metric_logger.log_every(data_loader, args.print_freq, header)):
         start_time = time.time()
         image, target = image.to(device), target.to(device)
-        with torch.cuda.amp.autocast(enabled=args.amp):
+        with torch.cuda.amp.autocast(enabled=scaler is not None):
```
Comment (prabhat00155) [inline references/classification/train.py:33]: Why this change? scaler is not None in case args.amp is true, but having args.amp here makes the code more readable.

File: torchvision/prototype/models/_api.py [hunk 2]
```diff
@@ -101,6 +101,11 @@ def get_weight(fn: Callable, weight_name: str) -> Weights:
         # TODO: Replace ann.__args__ with typing.get_args(ann) after python >= 3.8
         for t in ann.__args__:  # type: ignore[union-attr]
             if isinstance(t, type) and issubclass(t, Weights):
+                # ensure the name exists. handles builders with multiple types of weights like in quantization
+                try:
+                    t.from_str(weight_name)
+                except ValueError:
+                    continue
```
Comment (prabhat00155) [inline torchvision/prototype/models/_api.py:108]: Shouldn't we at least print a warning related to the exception rather than just ignoring it?

**Review by prabhat00155 — COMMENTED** [review]
Thanks @datumbox! I have a couple of comments.

File: references/classification/train.py (hunk 1 above)
Comment (datumbox) [reply references/classification/train.py:33]: This is only because in the quantized training script the method `train_one_epoch` is reused. Since there the `amp` param is not supported, accessing `args.amp` breaks the code. I think a more thorough refactoring is needed here but at least this fixes the broken code.

File: torchvision/prototype/models/_api.py (hunk 2 above)
Comment (datumbox) [reply torchvision/prototype/models/_api.py:108]: Ignoring is probably the right thing to do here. We try to match the provided string of the user with any known weight enums and if all fail, on the next couple of lines we throw an exception:
https://github.com/pytorch/vision/blob/015ee04a54908fc3ba9df65587d682f3d56268cf/torchvision/prototype/models/_api.py#L112-L115

Throwing a warning for every enum trying to match will only cause confusion.

**Review by prabhat00155 — APPROVED** [review]
